general/src/store/normalize.rs

```diff
@@ -132,18 +132,22 @@ impl<'a> NormalizeProcessor<'a> {
             });
         }
 
-        if event.server_name.value().is_some() {
+        let mut set_tag = |key, value| {
+            tags.retain(|tag| tag.value().and_then(|tag| tag.key()) != Some(key));
             tags.push(Annotated::new(TagEntry(
-                Annotated::new("server_name".to_string()),
-                std::mem::replace(&mut event.server_name, Annotated::empty()),
+                Annotated::new(key.to_string()),
+                value,
             )));
+        };
+
+        let server_name = std::mem::replace(&mut event.server_name, Annotated::empty());
+        if server_name.value().is_some() {
+            set_tag("server_name", server_name);
```

Comment:
This should become a method on `PairList` that doesn't remove and re-insert the value and instead replaces it.